src/Simplify.cpp

```diff
@@ -3908,6 +3924,68 @@ class Simplify : public IRMutator {
             IRMutator::visit(op);
         }
     }
+    template <typename T>
+    Expr hoist_slice_vector(Expr e) {
```

Comment:
Might be simpler if this just took Expr a, Expr b as args, instead of constructing a T at the call site, and then unpacking it again here.